### PR TITLE
[CIT-226] Document Thin Edge JSON format

### PR DESCRIPTION
**Pull request by albinsuresh**

### CODE REVIEW

File: docs/src/architecture/thin-edge-json.md [hunk 1]
```diff
@@ -1 +1,128 @@
...
+More topics will be added under the `tedge/` topic in future to support more data types like events, alarms etc.
+So it is advised to avoid any sub-topics under `tedge` for any other data exchange between processes.
+
+== Sending measurements to the cloud
+
+The `thin-edge.io` framework lets users forward any/all the measurements generated in the thin-edge device 
+to any IoT cloud provider that it is connected to, with the help of a *mapper* component designed for that cloud.
+The responsibility of a mapper is to subscribe to the `tedge/measurements` topic to receive all incoming measurements 
+represented in the cloud vendor neutral Thin Edge JSON format, to a format that the connected cloud understands.
+Refer to [Cloud Message Mapper Architecture](./mapper.md) for more details on the mapper component.
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply docs/src/architecture/thin-edge-json.md:128]: No, we are too short in time

File: docs/src/architecture/thin-edge-json.md [hunk 2]
```diff
@@ -1 +1,128 @@
...
+== Single-valued measurements
+
+Simple single-valued measurements like temperature or pressure measurement with a single value can be expressed as follows:
+
+```json
+{ "temperature": 25 }
+```
+
+where the key represents the measurement type and the value represents the measurement value.
+The keys can only have alphanumeric characters and the "_" (underscore) character but must not start with an underscore.
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply docs/src/architecture/thin-edge-json.md:16]: Your point is valid. The code has to be fixed. Let the doc unchanged.

**Review by didier-wenzek — APPROVED** [review]
Approved

File: docs/src/architecture/thin-edge-json.md [hunk 3]
```diff
@@ -1 +1,128 @@
...
+            "L2": 10.3
+        },
+        "phase3": {
+            "L3": 8.8
+        }
+    }
+}
+```
+
+because the values at the second level(`phase1`, `phase2` and `phase3`) are not numeric values.
```
[the remark this answers is not in the document]
Comment (makr11st) [reply docs/src/architecture/thin-edge-json.md:56]: Did we? This sounds like making the format pretty useless...

File: docs/src/architecture/thin-edge-json.md [hunk 4]
```diff
@@ -1 +1,128 @@
...
+Simple single-valued measurements like temperature or pressure measurement with a single value can be expressed as follows:
+
+```json
+{ "temperature": 25 }
+```
+
+where the key represents the measurement type and the value represents the measurement value.
+The keys can only have alphanumeric characters and the "_" (underscore) character but must not start with an underscore.
+The values can only be numeric.
+String or Boolean values are not allowed.
```
[the remark this answers is not in the document]
Comment (makr11st) [reply docs/src/architecture/thin-edge-json.md:18]: How about measurements like for example 'direction' where you may need to state actual direction let say N -> North or colour which may be a string? This change sounds like very limiting and not flexible. Even LWM2M supports strings and bools... http://www.openmobilealliance.org/wp/omna/lwm2m/lwm2mregistry.html

**Comment by albinsuresh** [comment]
Replaced with https://github.com/thin-edge/thin-edge.io/pull/141

**Pull request closed without merging**